pandas/indexes/multi.py

```diff
@@ -1907,6 +1907,7 @@ def convert_indexer(start, stop, step, indexer=indexer, labels=labels):
                 return np.array(labels == loc, dtype=bool)
             else:
                 # sorted, so can return slice object -> view
+                loc = labels.dtype.type(loc)
```

Comment:
Apparently not :-) 
But it's only a partial indexing test that fails. So `loc` could also be a slice, and it is expected in partial indexing that this raises an error in searchsorted, but now it raises already a slightly different error in `dtype.type(loc)`. So that can easily be solved with:

```
                try:
                    loc = labels.dtype.type(loc)
                except TypeError:
                    # this occurs when loc is a slice (partial string indexing)
                    # but the TypeError raised by searchsorted in this case
                    # is catched in Index._has_valid_type()
                    pass
```